tests/test_types.py

```
@@ -0,0 +1,433 @@
+# ── Interface / structural tests ────────────────────────────────────────────
+
+class TestPayloadInterface:
+    """Verify every payload class implements the ParsleyDataPayload interface."""
+
+    ALL_TYPES = [
+        GENERAL_BOARD_STATUS, RESET_CMD, DEBUG_RAW, CONFIG_SET, CONFIG_STATUS,
+        ACTUATOR_CMD, ACTUATOR_STATUS, ALT_ARM_CMD, ALT_ARM_STATUS,
+        SENSOR_ANALOG16, SENSOR_ANALOG32, SENSOR_DEM_ANALOG16,
+        GPS_TIMESTAMP, GPS_LATITUDE, GPS_LONGITUDE, GPS_ALTITUDE, GPS_INFO,
+        STREAM_STATUS, STREAM_DATA, STREAM_RETRY,
+    ]
```

Ruff (0.15.5)

[warning] Mutable default value for class attribute

(RUF012)